plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/manager/VmwareManagerImpl.java

```diff
@@ -1243,6 +1237,46 @@ public VmwareDatacenterVO addVmwareDatacenter(AddVmwareDcCmd cmd) throws Resourc
         return vmwareDc;
     }
 
+    VmwareDatacenterVO createOrUpdateDc(String guid, String name, String host, String user, String password) {
+        VmwareDatacenterVO vmwareDc = new VmwareDatacenterVO(guid, name, host, user, password);
+        // Add DC to database into vmware_data_center table
+        try {
+            vmwareDc = vmwareDcDao.persist(vmwareDc);
+        } catch (EntityExistsException e) {
+            // if that fails just get the record as is
+            vmwareDc = vmwareDcDao.getVmwareDatacenterByGuid(guid);
+            // we could now update:
+//            vmwareDc.setPassword(password);
+//            vmwareDc.setUser(user);
+//            vmwareDc.setVmwareDatacenterName(name);
+//            vmwareDc.setVcenterHost(host);
+//            vmwareDcDao.update(vmwareDc.getId(), vmwareDc);
+            // but let's assume user error for now
```

Comment:
Do we need this?
Shouldn't we first check for existing entry and create a new only if doesn't exist

Reply:
my point with this comment is that we could regard the new definition as an update (of more than just the guid)
btw this is not commented out code , but comment written as code. Not sure if that is a justification ;)

Reply:
what do you think @shwstppr

Reply:
I feel a single-line comment would also work. To me, this still feels like commented-out code

Reply:
```suggestion
            // we could now update the `vmwareDC` with the user supplied `password`, `user`, `name` and `host`,
            // but let's assume user error for now
```

I must say that I like my original comment better, but how about this, @shwstppr ?

Reply:
fine with me @DaanHoogland